haskell/cabal.bzl

```diff
@@ -86,15 +86,17 @@ def _cabal_tool_flag(tool):
         return "--with-{}={}".format(tool.basename, tool.path)
 
 def _make_path(hs, binaries):
-    return ":".join([binary.dirname for binary in binaries.to_list()])
+    path_list_sep = ";" if hs.toolchain.is_windows else ":"
+    return path_list_sep.join([binary.dirname for binary in binaries.to_list()])
 
 def _prepare_cabal_inputs(hs, cc, unix, dep_info, cc_info, component, package_id, tool_inputs, tool_input_manifests, cabal, setup, srcs, flags, cabal_wrapper, package_database):
     """Compute Cabal wrapper, arguments, inputs."""
     with_profiling = is_profiling_enabled(hs)
 
     (ghci_extra_libs, env) = get_ghci_extra_libs(hs, cc_info)
     env.update(**hs.env)
-    env["PATH"] = _make_path(hs, tool_inputs) + ":" + ":".join(unix.paths)
+    path_list_sep = ";" if hs.toolchain.is_windows else ":"
```

Comment:
nit: You can move this to the toplevel to avoid having to repeat it

Reply:
Good point, turns out this is a good opportunity to also refactor the path separator in `LD_LIBRARY_PATH`, so I'll do that in a separate PR.

Reply:
Done in https://github.com/tweag/rules_haskell/pull/1134